Week3/homework/7-step3.js

```diff
@@ -22,4 +22,12 @@ f2(y);
 console.log(y);
 
 // Add your explanation as a comment here
+/* 
+In Javascript all types of data except objects define immutable values (values which are incapable of being changed).
+We refer to values of these types as "primitive values".
+In the case of objects, JavaScript always uses the Object by reference when it passes as argument, so if we pass objects to a function and call the Function, the function updates the value of the object.
+In our first case, the type of data that is assigned to x is a number, which is a primitive value.
+In case we want to console.log the output of the first function, we must console.log(f1(x)) instead. 
+In our second case, the type of data that is assigned to x is an object (not a primitive value). 
+*/
```

Comment:
Technically correct, it might also be wise to mention that an Array is an object, so the some rule holds for them.  Otherwise people think that objects and arrays are different.